src/Credentials/GCECredentials.php

```diff
@@ -126,28 +139,32 @@ public static function onGce(callable $httpHandler = null)
             $httpHandler = HttpHandlerFactory::build();
         }
         $checkUri = 'http://' . self::METADATA_IP;
-        try {
-            // Comment from: oauth2client/client.py
-            //
-            // Note: the explicit `timeout` below is a workaround. The underlying
-            // issue is that resolving an unknown host on some networks will take
-            // 20-30 seconds; making this timeout short fixes the issue, but
-            // could lead to false negatives in the event that we are on GCE, but
-            // the metadata resolution was particularly slow. The latter case is
-            // "unlikely".
-            $resp = $httpHandler(
-                new Request('GET', $checkUri),
-                ['timeout' => 0.3]
-            );
-
-            return $resp->getHeaderLine(self::FLAVOR_HEADER) == 'Google';
-        } catch (ClientException $e) {
-            return false;
-        } catch (ServerException $e) {
-            return false;
-        } catch (RequestException $e) {
-            return false;
+        for ($i = 1; $i <= self::MAX_COMPUTE_PING_TRIES; $i++) {
+            try {
+                // Comment from: oauth2client/client.py
+                //
+                // Note: the explicit `timeout` below is a workaround. The underlying
+                // issue is that resolving an unknown host on some networks will take
+                // 20-30 seconds; making this timeout short fixes the issue, but
+                // could lead to false negatives in the event that we are on GCE, but
+                // the metadata resolution was particularly slow. The latter case is
+                // "unlikely".
+                $resp = $httpHandler(
+                    new Request('GET', $checkUri),
+                    ['timeout' => self::COMPUTE_PING_CONNECTION_TIMEOUT_S]
+                );
+
+                return $resp->getHeaderLine(self::FLAVOR_HEADER) == 'Google';
+            } catch (ClientException $e) {
+                // error_log("ClientException", 0);
+            } catch (ServerException $e) {
+                // error_log("ServerException", 0);
+            } catch (RequestException $e) {
+                // error_log("RequestException", 0);
+            }
+            $httpHandler = HttpHandlerFactory::build();
```

Comment:
Need to build a new httpHandler instead of using the old one, otherwise it will throw `OUTBOUND Exception`
https://travis-ci.org/google/google-auth-library-php/builds/388429882?utm_source=github_status&utm_medium=notification

However, if `$httpHandler` is not null and passed into this function. Should we make a copy of that here?

Reply:
I'm ok with the current approach

Reply:
Thanks!